x/wasm/internal/keeper/proposal_handler.go

```diff
@@ -93,7 +93,15 @@ func handleMigrateProposal(ctx sdk.Context, k Keeper, p types.MigrateContractPro
 		sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
 		sdk.NewAttribute(types.AttributeKeyContract, p.Contract.String()),
 	)
-	ctx.EventManager().EmitEvents(append(res.Events, ourEvent))
+	ctx.EventManager().EmitEvent(ourEvent)
+
+	for _, e := range res.Events {
```

Comment:
What is going on here>

res.Events.Attributes are in some abci format (`[]byte`?) which is different than the `sdk.Attribute`, so we grab them and convert them to strings for the event manager? (Which then converts them back to bytes to emit them likely)

Reply:
Yes 🙈